src/agreement/agreement-pool.ts

```diff
@@ -85,8 +85,17 @@ export class AgreementPool {
     this.events.emit("destroyed", agreement.getDto());
```

Comment:
this event should be emitted from the function in createFactoryPool because it will not be emitted here if the agreemnet is destroyed during the `drainAndClear` process